plugins/reach/services/EntryVendorTaskService.php

```diff
@@ -0,0 +1,356 @@
+			throw new KalturaAPIException(KalturaReachErrors::CANNOT_REJECT_NOT_MODERATED_TASK);
+		
+		$dbEntryVendorTask->setModeratingUser($this->getKuser()->getPuserId());
+		$dbEntryVendorTask->setStatus(KalturaEntryVendorTaskStatus::REJECTED);
+		$dbEntryVendorTask->save();
+		
+		// return the saved object
+		$entryVendorTask = new KalturaEntryVendorTask();
+		$entryVendorTask->fromObject($dbEntryVendorTask, $this->getResponseProfile());
+		return $entryVendorTask;
+	}
+	
+	/**
+	 * get KalturaEntryVendorTask objects for specific vendor partner
+	 *
+	 * @action getJobs
+	 * @param KalturaEntryVendorTaskFilter $filter
+	 * @param KalturaFilterPager $pager
+	 * @return KalturaEntryVendorTaskListResponse
+	 */
+	public function getJobsAction(KalturaEntryVendorTaskFilter $filter = null, KalturaFilterPager $pager = null)
+	{
+		if (!PermissionPeer::isValidForPartner(PermissionName::REACH_VENDOR_PARTNER_PERMISSION, kCurrentContext::getCurrentPartnerId()))
+			throw new KalturaAPIException(KalturaReachErrors::ENTRY_VENDOR_TASK_SERVICE_GET_JOB_NOT_ALLOWED, kCurrentContext::getCurrentPartnerId());
+		
+		if (!$filter)
+			$filter = new KalturaEntryVendorTaskFilter();
+		
+		$filter->vendorPartnerIdEqual = kCurrentContext::getCurrentPartnerId();
+		$filter->statusEqual = EntryVendorTaskStatus::PENDING;
+		if (!$pager)
+			$pager = new KalturaFilterPager();
+		
+		return $filter->getListResponse($pager, $this->getResponseProfile());
+	}
+	
+	/**
+	 * Update entry vendor task. Only the properties that were set will be updated.
+	 *
+	 * @action updateJob
+	 * @param int $id vendor task id to update
+	 * @param KalturaEntryVendorTask $entryVendorTask evntry vendor task to update
+	 * @return KalturaEntryVendorTask
+	 * @throws KalturaReachErrors::ENTRY_VENDOR_TASK_NOT_FOUND
+	 */
+	public function updateJobAction($id, KalturaEntryVendorTask $entryVendorTask)
+	{
+		if (!PermissionPeer::isValidForPartner(PermissionName::REACH_VENDOR_PARTNER_PERMISSION, kCurrentContext::getCurrentPartnerId()))
+			throw new KalturaAPIException(KalturaReachErrors::ENTRY_VENDOR_TASK_SERVICE_GET_JOB_NOT_ALLOWED, kCurrentContext::getCurrentPartnerId());
+		
+		$dbEntryVendorTask = EntryVendorTaskPeer::retrieveByPKAndVendorPartnerId($id, kCurrentContext::getCurrentPartnerId());
+		if (!$dbEntryVendorTask)
+			throw new KalturaAPIException(KalturaReachErrors::ENTRY_VENDOR_TASK_NOT_FOUND, $id);
+		
+		$dbEntryVendorTask = $entryVendorTask->toUpdatableObject($dbEntryVendorTask);
+		$dbEntryVendorTask->save();
+		
+		// return the saved object
+		$entryVendorTask = new KalturaEntryVendorTask();
+		$entryVendorTask->fromObject($dbEntryVendorTask, $this->getResponseProfile());
+		return $entryVendorTask;
+	}
+	
+	/**
+	 * Cancel entry task. will only occur for task in PENDING or PENDING_MODERATION status
+	 *
+	 * @action abort
+	 * @param int $id vendor task id
+	 * @param string $abortReason
+	 * @return KalturaEntryVendorTask
+	 * @throws KalturaReachErrors::ENTRY_VENDOR_TASK_NOT_FOUND
```

Comment:
Better be protected using @validateUser annotation, since any user KS can call this api.